pallets/stateful-storage/src/signed_extension.rs

```diff
@@ -0,0 +1,200 @@
+//! Substrate Signed Extension for validating requests to the stateful storage pallet
+use crate::{Call, Config, Error, Pallet};
+use common_primitives::{
+	msa::{MessageSourceId, MsaValidator, SchemaId},
+	stateful_storage::{PageHash, PageId},
+};
+use core::marker::PhantomData;
+use frame_support::{
+	dispatch::DispatchInfo, ensure, pallet_prelude::ValidTransaction, traits::IsSubType,
+};
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{DispatchInfoOf, Dispatchable, SignedExtension},
+	transaction_validity::{InvalidTransaction, TransactionValidity, TransactionValidityError},
+	DispatchError,
+};
+
+/// The SignedExtension trait is implemented on CheckFreeExtrinsicUse to validate the request. The
+/// purpose of this is to ensure that the target_hash is verified in transaction pool before getting
+/// into block. This is to reduce the chance that capacity consumption due to stale hash
```

Comment:
nit. only if you are already going to change something in this file
```suggestion
/// into block. This is to reduce the chance of capacity consumption due to stale hash
```